app.js

```diff
@@ -41,37 +40,51 @@ addMushroomButton.addEventListener('click', () => {
 });
 
 addFriendButton.addEventListener('click', () => {
-    // get the name from the input
+    let newFriends = friendInputEl.value;
     // create a new friend object
-    // push it into the friends state array, passed in as an argument
+    const newFriend = {
+        name: newFriends,
+        satisfaction: 1,
+    };
+
+    friendData.push(newFriend);
     // reset the input
+    friendInputEl.value = '';
     // display all the friends (use a function here)
+    displayFriends();
 });
 
 function displayFriends() {
     // clear out the friends in DOM
-
+    friendsEl.textContent = '';
     // for each friend in state . . .
     for (let friend of friendData) {
         // use renderFriend to make a friendEl
-
+        const friendEl = renderFriend(friend);
         // this is a clickable list, so . . .
-        //     add an event listener to each friend
-        //         and if the friend's satisfaction level is below 3 and you have mushrooms left
-        //             increment the friends satisfaction and decrement your mushrooms
-        //             then display your friends and mushrooms with the updated state
+        friendEl.addEventListener('click', () => {
+            if (friend.satisfaction < 3 && mushroomCount > 0) {
+                friend.satisfaction++;
+                mushroomCount--;
+            }
```

Comment:
You can add a conditional in this event listener that alerts the user to go hunting for more mushrooms if the mushroomCount === 0!